doc/python/performance.md

```diff
@@ -0,0 +1,298 @@
+    file_extension: .py
+    mimetype: text/x-python
+    name: python
+    nbconvert_exporter: python
+    pygments_lexer: ipython3
+    version: 3.11.10
+  plotly:
+    description: Recommendations for increased speed, improved interactivity, and
+      the ability to plot even more data!
+    display_as: basic
+    language: python
+    layout: base
+    name: High Performance Visualization
+    order: 14
+    permalink: python/performance/
+    redirect_from: 
+    - python/webgl-vs-svg/
+    - python/datashader/
+    thumbnail: thumbnail/webgl.jpg
+---
+
+## WebGL
+
+`plotly` figures are rendered by web browsers, which broadly speaking have two families of capabilities for rendering graphics:
+
+- The SVG API, which supports vector rendering.
+- The Canvas API, which supports raster rendering, and can exploit GPU hardware acceleration via a browser technology known as WebGL.
+
+Each `plotly` trace type is rendered with either SVG or WebGL. The following trace types use WebGL for rendering:
+
+* Accelerated versions of SVG trace types: `scattergl`, `scatterpolargl`,
+* High-performance multidimensional trace types: `splom`, or `parcoords`
+* 3-d trace types `scatter3d`, `surface`, `mesh3d`, `cone`, `streamtube`
+* Mapbox Gl JS-powered trace types: `scattermapbox`, `choroplethmapbox`, `densitymapbox`
+
+### WebGL Limitations and Tradeoffs
+
+WebGL is a powerful technology for accelerating computation but comes with some strict limitations:
+
+1. GPU requirement: WebGL is a GPU (graphics card) technology and therefore requires specific hardware which is available in most but not all cases and is supported by most but not all browsers.
+2. Rasterization: WebGL-rendered data is drawn as a grid of pixels rather than as individual shapes, so can appear pixelated or fuzz in certain cases, and when exported to static file formats will appear pixelated on zoom. In addition: text rendering will differ between SVG and WebGL-powered traces.
+3. Context limits: browsers impose a strict limit on the number of WebGL "contexts" that any given web document can access. WebGL-powered traces in `plotly` can use multiple contexts in some cases but as a general rule, **it may not be possible to render more than 8 WebGL-involving figures on the same page at the same time.** See the following section, Multiple WebGL Contexts, for more details.
+4. Size limits: browsers impose hardware-dependent limits on the height and width of figures using WebGL which users may encounter with extremely large plots (e.g. tens of thousands of pixels of height).
+
+In addition to the above limitations, the WebGL-powered version of certain SVG-powered trace types (`scattergl`, `scatterpolargl`) are not complete drop-in replacements for their SVG counterparts yet
+* Available symbols will differ.
+* Area fills are not yet supported in WebGL.
+* Range breaks on time-series axes are not yet supported.
+* Axis range heuristics may differ.
+
+#### Multiple WebGL Contexts
+
+*New in 5.19*
```

Comment:
is this relevant for 6.0 ?

Reply:
Yeah, just keeping the 5.19 in so users that are using an older version will know if this is available to them